Assets/Scripts/Agents/IAgent.cs

```diff
@@ -13,18 +13,23 @@ public interface IAgent {
   IMovement Movement { get; set; }
   IController Controller { get; set; }
   ISensor Sensor { get; set; }
+  IAgent TargetModel { get; set; }
```

Comment:
Documentation needed
```suggestion
  // Target model maintained by the agent's sensor. Represents the agent's best estimate
  // of the target's state based on sensor measurements and prediction.
  // Managed by HierarchicalAgent when targets are assigned/unassigned.
  IAgent TargetModel { get; set; }
```